requirements.txt

```
@@ -12,7 +12,7 @@ ninja>=1.10.0.post2
 pybind11>=2.10.1
 pycnite>=2024.07.31
 pydot>=1.4.2
-pylint>=2.14.4,<3.2.4  # https://github.com/pylint-dev/pylint/issues/9751
+pylint>=3.0.0,!=3.2.4 # Python 3.12 support was officially introduced in pylint 3.0.0 (Python 3.7 support, which we don't need, was officially dropped) https://pylint.pycqa.org/en/stable/whatsnew/3/3.0/index.html  # 3.2.4 was a bad version I guess: https://github.com/pylint-dev/pylint/issues/9751
```

I'm going to remove the comment from here, it's better suited for the description of the pull request

```suggestion
pylint>=3.0.0,!=3.2.4  # https://github.com/pylint-dev/pylint/issues/9751
```